x/bank/keeper/keeper.go

```diff
@@ -412,6 +417,58 @@ func (k BaseKeeper) BurnCoins(ctx context.Context, moduleName string, amounts sd
 	return nil
 }
 
+// SubtractCoins subtracts coins from the balance of an account
+// NOTE(Heimdall-v2): required in heimdall for fee withdrawal in topup module
+func (k BaseKeeper) SubtractCoins(ctx context.Context, moduleName string, addr sdk.AccAddress, amounts sdk.Coins) error {
```

Comment:
This seems like a potential footgun, what is the use case for SubtractCoins? If it's part of an atomic action to subtract coins from one account to send to another than those already exist. This method also emits a burn event, so then why not just send coins to a module account and burn them?

Reply:
So as the comment above suggests, the `topup` module in heimdall makes use of this for [fee withdrawal from a validator's account](https://github.com/maticnetwork/heimdall/blob/249aa798c2f23c533d2421f2101127c11684c76e/topup/handler.go#L90). 
Doing this in two steps was an option I contemplated but I still decided to keep it for the time being for compatibility. Do you see any attack vector in this ?
Also I realise the  `SubtractCoins` doesn't need a `moduleName` param.

Reply:
This is the same steps as a burn function though. It doesn't make sense to add this function when burn is removing coins from total supply. This function is even emitting a burn event.

Reply:
I believe I removed it in this commit: [1cbb41b](https://github.com/0xPolygon/cosmos-sdk/pull/5/commits/1cbb41b4cb73390c58ced14d1e5f76288a1802a3)